eclair-core/src/main/scala/fr/acinq/eclair/blockchain/watchdogs/HeadersOverDns.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.blockchain.watchdogs
+
+import akka.actor.Status
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.adapter.TypedActorRefOps
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.io.dns.{AAAARecord, DnsProtocol}
+import akka.io.{Dns, IO}
+import fr.acinq.bitcoin.{Block, BlockHeader, ByteVector32}
+import fr.acinq.eclair.blockchain.watchdogs.BlockchainWatchdog.BlockHeaderAt
+import fr.acinq.eclair.blockchain.watchdogs.Monitoring.{Metrics, Tags}
+import org.slf4j.Logger
+import scodec.bits.BitVector
+
+/**
+ * Created by t-bast on 29/09/2020.
+ */
+
+/** This actor queries https://bitcoinheaders.net/ to fetch block headers. */
+object HeadersOverDns {
+
+  val Source = "bitcoinheaders.net"
+
+  // @formatter:off
+  sealed trait Command
+  case class CheckLatestHeaders(replyTo: ActorRef[BlockchainWatchdog.LatestHeaders]) extends Command
+  private case class WrappedDnsResolved(response: DnsProtocol.Resolved) extends Command
+  private case class WrappedDnsFailed(cause: Throwable) extends Command
+  // @formatter:on
+
+  def apply(chainHash: ByteVector32, currentBlockCount: Long): Behavior[Command] = {
```

Comment:
I think you are _almost_ at the point where it makes sense to use the [private class](https://doc.akka.io/docs/akka/current/typed/style-guide.html#passing-around-too-many-parameters) pattern.

Reply:
I'll look into it

Reply:
See `ChannelRelay`/`NodeRelay` for inspiration 😄

Reply:
TBH I think I don't have that many arguments, so let's keep it like that :)

Reply:
Fair enough.